app/src/androidTest/java/org/simple/clinic/encounter/EncounterRepositoryAndroidTest.kt

```diff
@@ -99,4 +102,106 @@ class EncounterRepositoryAndroidTest {
       assertThat(updatedAt).isEqualTo(Instant.parse("2018-02-13T00:00:00Z"))
     }
   }
+
+  @Test
+  fun when_a_bp_is_saved_encounter_should_get_created() {
+    //given
+    val patientUuid = UUID.fromString("6f725ffd-7008-4018-9e7c-33346964a0c1")
+    val facilityUuid = UUID.fromString("22bdeedb-061e-4a17-8739-e946a4206593")
+    val bpUuid = UUID.fromString("2edd4c06-e2de-4a18-a8e7-43e2e30c9aba")
+    val encounteredDate = LocalDate.parse("2018-01-01")
+    val bloodPressureMeasurement = testData.bloodPressureMeasurement(
+        uuid = bpUuid,
+        patientUuid = patientUuid,
+        facilityUuid = facilityUuid,
+        recordedAt = Instant.parse("2018-01-01T00:00:00Z"),
+        createdAt = Instant.parse("2018-01-01T00:00:00Z"),
+        updatedAt = Instant.parse("2018-01-01T00:00:00Z"),
+        syncStatus = SyncStatus.PENDING,
+        encounterUuid = generateEncounterUuid(facilityUuid, patientUuid, encounteredDate)
+    )
+
+    //when
+    repository.saveBloodPressureMeasurement(bloodPressureMeasurement, encounteredDate).blockingAwait()
+
+    //then
+    val bps = appDatabase.bloodPressureDao().bloodPressure(bpUuid).blockingFirst()
+    val encounters = appDatabase.encountersDao().recordsWithSyncStatus(SyncStatus.PENDING).blockingFirst()
+
+    assertThat(encounters).isNotEmpty()
+
+    with(encounters.first()) {
```

Comment:
This enclosing `with` block makes it harder to comprehend what is going on. Can we split this into two separate assertion blocks?